src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationImporter.java

```diff
@@ -2,25 +2,40 @@
 
 import org.mskcc.cbio.portal.dao.DaoException;
 import org.mskcc.cbio.portal.dao.DaoGeneticAlteration;
+import org.mskcc.cbio.portal.dao.DaoGeneticProfileSamples;
 import org.mskcc.cbio.portal.model.CanonicalGene;
 import org.mskcc.cbio.portal.util.ProgressMonitor;
 
-import java.util.*;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 import static java.lang.String.format;
 
 public class GeneticAlterationImporter {
 
-    private final int geneticProfileId;
-    private Set<Long> importSetOfGenes = new HashSet<>();
-    private DaoGeneticAlteration daoGeneticAlteration;
+    protected int geneticProfileId;
+    protected List<Integer> orderedSampleList;
+    private final Set<Long> importSetOfGenes = new HashSet<>();
+    private final Set<Integer> importSetOfGeneticEntityIds = new HashSet<>();
 
+    private final DaoGeneticAlteration daoGeneticAlteration = DaoGeneticAlteration.getInstance();
+
+    protected GeneticAlterationImporter() {}
     public GeneticAlterationImporter(
-        int geneticProfileId, 
-        DaoGeneticAlteration daoGeneticAlteration
-    ) {
+        int geneticProfileId,
+        List<Integer> orderedSampleList
+    ) throws DaoException {
         this.geneticProfileId = geneticProfileId;
-        this.daoGeneticAlteration = daoGeneticAlteration;
+        this.orderedSampleList = orderedSampleList;
+        storeOrderedSampleList();
```

Comment:
This makes construction of this class dependent on a successful database operation. In general I think this is not recommended. Could this function call be moved inside of the store() methods below? Also, if the storage of the sample id list succeeds here, but the storage of the cna event profile fails during store() would we have a problem? (or maybe transactions will take care of that?)

I suppose that each instance of this class is specific to a particular genetic alteration profile, so we do not need to worry about reuse of any instance to import a different profile (and maybe need to store a different set of sample ids)

Reply:
> Could this function call be moved inside of the store() methods below?
we should call `storeOrderedSampleList()` once only. It's possible to have a flag and do this operation on first call of the `store()` method.

> if the storage of the sample id list succeeds here, but the storage of the cna event profile fails during store() would we have a problem? (or maybe transactions will take care of that?)
this code should run in a single transaction indeed.

Reply:
how about having an explicit `initialise()` method that the client side of the code has to call?  https://github.com/cBioPortal/cbioportal-core/pull/48/commits/0edaec3fc67d580dd7ebd30913c50f8ea989de0f

Reply:
Yes, the initialise (english spelling) call is good I think. Sometimes in cases like this, rather than depending on the caller/client code to call in to initialize() before using the functionality, there is a data member which can be used to determine whether or not the initialization has occurred yet or not. Then there can be a conditional initialization ... a test in the actual business logic which asks (is this initialized yet? ... if not yet then call initialize()). That way the initialization cannot be erroneously skipped in the client code.

But this looks good to me.